eras/conway/impl/src/Cardano/Ledger/Conway/Core.hs

```diff
@@ -66,6 +66,8 @@ class BabbageEraTxBody era => ConwayEraTxBody era where
   -- | Lens for getting and setting `ProposalProcedures`.
   proposalProceduresTxBodyL :: Lens' (TxBody era) (StrictSeq (ProposalProcedure era))
 
+  treasuryDonationTxBodyL :: Lens' (TxBody era) Coin
```

Comment:
I think it is better to have an explicit information about presence of this field in the transaction, instead of relying on this value to be 0 when no donation happens. In fact, I would even go as far as disallowing 0 amount at the serialization level or with a predicate check.
```suggestion
  treasuryDonationTxBodyL :: Lens' (TxBody era) (StrictMaybe Coin)
```

Reply:
Why so? It seems to me it would only complicate the code.

Reply:
Because the intention becomes explicit.

Supplying `0` and not supplying field at all should not mean the same thing.

Reply:
As per our discussion, let's prevent zeros on the wire and we can get rid of the Maybe